test/e2e/custom_resource_definition.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	//"encoding/json"
+	//"fmt"
+	//"net/http"
+	//"time"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apiextensions-apiserver/test/integration/testserver"
+	//"k8s.io/apimachinery/pkg/api/meta"
+	//metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/apimachinery/pkg/runtime"
+	//"k8s.io/apimachinery/pkg/util/wait"
+	//"k8s.io/apimachinery/pkg/runtime/schema"
+	//"k8s.io/kubernetes/pkg/api"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+/*
+type Noxu struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" description:"standard object metadata"`
+
+	SomeField  string `json:"someField"`
+	OtherField int    `json:"otherField"`
+}
+
+type NoxuList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" description:"standard list metadata; see http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata"`
+
+	Items []Noxu `json:"items"`
+}
+
+func createCustomResourceDefinition(crd *v1beta1.CustomResourceDefinition, apiExtensionClient clientset.Interface, clientPool dynamic.ClientPool) (*dynamic.Client, func(), error) {
```

Comment:
@deads2k I was trying to write a function to create CDR, return defer and dynamic client like you suggested, but turn out I just found it mostly doing the same job with `testserver.CreateNewCustomResourceDefinition` since after CDR created in this function it also have some `List` `Create` `Delete` operation, so do you think I can just re-use those functions in integration tests ?